components/connector-service/internal/monitoring/monitoring.go

```diff
@@ -0,0 +1,62 @@
+package monitoring
+
+import (
+	"github.com/kyma-project/kyma/components/connector-service/internal/apperrors"
+	"github.com/kyma-project/kyma/components/connector-service/internal/monitoring/collector"
+	"github.com/kyma-project/kyma/components/connector-service/internal/monitoring/middleware"
+	"github.com/prometheus/client_golang/prometheus"
+	"net/http"
+)
+
+const (
+	calculatedPercentile = 99.5
+	percentileAccuracy   = 0.05
+)
+
+type Middleware interface {
+	Handle(next http.Handler) http.Handler
```

Comment:
Do we need this interface? SetupMonitoring. We could:
1) define type MiddlewareFunction
type  MiddlewareFunc func (next http.Handler) http.Handler
2) change SetupMonitoring to return []MiddlewareFunc

We could even inline function definition as follows:
func SetupMonitoring() ([]func (next http.Handler) http.Handler, apperrors.AppError)

Reply:
I have used MiddlewareFunc from mux package, which solves the problem